frame/dapps-staking/src/tests.rs

```diff
@@ -1059,6 +1059,71 @@ fn unbond_and_unstake_with_no_chunks_allowed() {
     })
 }
 
+#[test]
+fn rebond_and_stake_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+        let rebond_contract_id = MockSmartContract::Evm(H160::repeat_byte(0x02));
+        assert_register(10, &contract_id);
+        assert_register(11, &rebond_contract_id);
+
+        let staker_id = 1;
+        assert_bond_and_stake(staker_id, &contract_id, 1000);
+
+        let first_unbond_value = 100;
+        let second_unbond_value = 250;
+        let initial_era = DappsStaking::current_era();
+
+        // Unbond some amount in the initial era
+        assert_unbond_and_unstake(staker_id, &contract_id, first_unbond_value);
+
+        // Advance one era and then unbond some more
+        advance_to_era(initial_era + 1);
+        assert_unbond_and_unstake(staker_id, &contract_id, second_unbond_value);
+
+        // rebond and stake
+        assert_rebond_and_stake(staker_id, &rebond_contract_id);
+    })
+}
+
+#[test]
+fn rebond_and_stake_unexist_contract_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let staker_id = 1;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+        assert_register(10, &contract_id);
+
+        assert_bond_and_stake(staker_id, &contract_id, 1000);
+        assert_unbond_and_unstake(staker_id, &contract_id, 100);
+
+        let non_exist_contract_id = MockSmartContract::Evm(H160::repeat_byte(0x02));
+        assert_noop!(
+            DappsStaking::rebond_and_stake(Origin::signed(staker_id), non_exist_contract_id),
+            Error::<TestRuntime>::NotOperatedContract,
+        );
+    })
+}
+
+#[test]
+fn rebond_and_stake_no_unbonding_chunks_fails() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let staker_id = 1;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+        assert_register(10, &contract_id);
+
+        assert_noop!(
+            DappsStaking::rebond_and_stake(Origin::signed(staker_id), contract_id),
+            Error::<TestRuntime>::NothingToRebond,
+        );
+    })
+}
+
```

Comment:
I'd suggest also covering the error cases with InsufficientValue or TooManyEraStakeValues.

Not mandatory, just a suggestion.